src/views/NewArticle.js

```diff
@@ -0,0 +1,195 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import RichTextEditor from "react-rte";
+import Input from "../components/common/Inputs/TextInput";
+import Button from "../components/common/Buttons/confirm";
+import Tag from "../components/common/Tags/NewTag";
+import { toolbarConfig } from "../components/common/editor/editorConfig";
+import {
+  handleInputField,
+  newArticle,
+  handleCreateTag,
+  removeTag
+} from "../redux/actions/newArticle";
+
+export const mapDispatchToProps = dispatch => ({
+  onInputChange: field => dispatch(handleInputField(field)),
+  newTag: tag => dispatch(handleCreateTag(tag)),
+  createArticle: article => dispatch(newArticle(article)),
+  handleRemoveTag: tag => dispatch(removeTag(tag))
+});
+
+const mapStateToProps = state => ({
+  response: state.article.response,
+  article: state.article
+});
+
+export class CreateArticle extends Component {
+  constructor() {
+    super();
+    this.state = {
+      tag: "",
+      error: "",
+      value: RichTextEditor.createEmptyValue()
+    };
+    this.submitArticle = this.submitArticle.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.onEditorChange = this.onEditorChange.bind(this);
+    this.onTagInputChange = this.onTagInputChange.bind(this);
+    this.handleNewTag = this.handleNewTag.bind(this);
+    this.onRemoveTag = this.onRemoveTag.bind(this);
+  }
+
+  onEditorChange(value) {
+    this.setState({ value });
+    const { onInputChange } = this.props;
+    onInputChange({ name: "body", value: value.toString("html") });
+  }
+
+  onTagInputChange(event) {
+    this.setState({
+      tag: event.target.value
+    });
+  }
+
+  onRemoveTag(value) {
+    const { handleRemoveTag } = this.props;
+    handleRemoveTag(value);
+  }
+
+  handleInputChange(event) {
+    const { name } = event.target;
+    const { value } = event.target;
+    const { onInputChange } = this.props;
+    onInputChange({ name, value });
+  }
+
+  submitArticle() {
+    const { article } = this.props;
+    const { title, body, description, tagsList } = article;
+    const { createArticle } = this.props;
+
+    if (!title || title.length < 10) {
```

Comment:
consider moving all validation into their own file under utils. We should keep our component as minimal as possible.